py/dml/dmlc.py

```diff
@@ -558,8 +584,32 @@ def main(argv):
               + "The DMLC developers WILL NOT respect their use. "
               + "NEVER enable this flag for any kind of production code!!!***")
 
-    dml.globals.api_version = options.simics_api
+    dml.globals.api_version = api_map[options.simics_api]
```

Comment:
Move this up to right after the `if options.simics_api not in api_map:` check, and adapt all old code that check `options.simics_api` to use `dml.globals.api_version` instead (bound to a `api_version` variable if spelling it out is too lengthy).

Reply:
Did an overhaul of all API handling